apps/full-site-editing/full-site-editing-plugin/donations/index.js

```diff
@@ -0,0 +1,101 @@
+/* eslint-disable wpcalypso/import-docblock */
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { getBlockType, registerBlockType, unregisterBlockType } from '@wordpress/blocks';
+import { __, _x } from '@wordpress/i18n';
+/* eslint-enable wpcalypso/import-docblock */
+
+/**
+ * Internal dependencies
+ */
+import edit from './edit';
+import './style.scss';
+
+/**
+ * Appends a "paid" tag to the Donations block title if site requires an upgrade.
+ */
+const addPaidBlockFlags = async () => {
+	let membershipsStatus;
+	try {
+		membershipsStatus = await apiFetch( { path: '/wpcom/v2/memberships/status' } );
+	} catch {
+		// Do not add any flag if request fails.
+		return;
+	}
+	const shouldUpgrade = membershipsStatus.should_upgrade_to_access_memberships;
+	if ( shouldUpgrade ) {
+		const donationsBlock = getBlockType( 'a8c/donations' );
+		if ( ! donationsBlock ) {
+			return;
+		}
+
+		const paidFlag = _x(
+			'paid',
+			'Short label appearing near a block requiring a paid plan',
+			'premium-content'
+		);
+
+		unregisterBlockType( 'a8c/donations' );
+		registerBlockType( 'a8c/donations', {
+			...donationsBlock,
+			title: `${ donationsBlock.title } (${ paidFlag })`,
+		} );
+	}
+};
+
+function registerDonationsBlock() {
+	registerBlockType( 'a8c/donations', {
+		title: __( 'Donations (a8c-only)', 'donations' ),
+		description: __( 'Accept donations on your site.', 'donations' ),
```

Comment:
I assumed this is based in the premium content block where we were incorrectly using the `premium-content` text domain. See #42583.
```suggestion
		title: __( 'Donations (a8c-only)', 'full-site-editing' ),
		description: __( 'Accept donations on your site.', 'full-site-editing' ),
```